sdk/identity/identity/src/credentials/azurePipelinesCredential.ts

```diff
@@ -113,41 +114,51 @@ export class AzurePipelinesCredential implements TokenCredential {
       }),
     });
     const response = await this.identityClient.sendRequest(request);
-    const text = response.bodyAsText;
-    if (!text) {
-      logger.error(
-        `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
-          response.status
-        }. Complete response - ${JSON.stringify(response)}`,
-      );
-      throw new AuthenticationError(
-        response.status,
-        `${credentialName}: Authenticated Failed. Received null token from OIDC request. Response status- ${
-          response.status
-        }. Complete response - ${JSON.stringify(response)}`,
-      );
-    }
-    try {
-      const result = JSON.parse(text);
-      if (result?.oidcToken) {
-        return result.oidcToken;
-      } else {
-        let errorMessage = `${credentialName}: Authentication Failed. oidcToken field not detected in the response.`;
-        if (response.status !== 200) {
-          errorMessage += `Response = ${JSON.stringify(result)}`;
-        }
-        logger.error(errorMessage);
-        throw new AuthenticationError(response.status, errorMessage);
+    return handleOidcResponse(response);
+  }
+}
+
+export async function handleOidcResponse(response: PipelineResponse): Promise<string> {
```

Comment:
We discussed making this function synchronous 👍

Reply:
done :)